applications/StructuralMechanicsApplication/custom_elements/membrane_element.cpp

```diff
@@ -0,0 +1,1292 @@
+
+namespace Kratos
+{
+
+// Constructor
+MembraneElement::MembraneElement( IndexType NewId, GeometryType::Pointer pGeometry )
+    : Element( NewId, pGeometry )
+{
+
+}
+
+// Constructor
+MembraneElement::MembraneElement(IndexType NewId, GeometryType::Pointer pGeometry, PropertiesType::Pointer pProperties)
+    : Element( NewId, pGeometry, pProperties )
+{
+
+}
+
+//***********************************************************************************
+//***********************************************************************************
+
+Element::Pointer MembraneElement::Create(
+    IndexType NewId,
+    NodesArrayType const& rThisNodes,
+    PropertiesType::Pointer pProperties) const
+
+{
+    return Kratos::make_intrusive< MembraneElement >(NewId, GetGeometry().Create(rThisNodes), pProperties);
+}
+
+//***********************************************************************************
+//***********************************************************************************
+
+Element::Pointer MembraneElement::Create(
+    IndexType NewId,
+    GeometryType::Pointer pGeom,
+    PropertiesType::Pointer pProperties) const
+
+{
+    return Kratos::make_intrusive< MembraneElement >(NewId, pGeom, pProperties);
+}
+
+
+//***********************************************************************************
+//***********************************************************************************
+
+void MembraneElement::EquationIdVector(
+    EquationIdVectorType& rResult,
+    ProcessInfo& rCurrentProcessInfo)
+
+{
+  KRATOS_TRY;
+
+  SizeType num_nodes, local_size;
+  SizeType local_index = 0;
+
+  num_nodes = GetGeometry().size();
+  local_size = num_nodes * 3;
+
+  const SizeType d_pos = this->GetGeometry()[0].GetDofPosition(DISPLACEMENT_X);
+
+  if (rResult.size() != local_size)
+      rResult.resize(local_size, false);
+
+  for (SizeType i_node = 0; i_node < num_nodes; ++i_node)
+  {
+      rResult[local_index++] = this->GetGeometry()[i_node].GetDof(DISPLACEMENT_X, d_pos).EquationId();
+      rResult[local_index++] = this->GetGeometry()[i_node].GetDof(DISPLACEMENT_Y, d_pos + 1).EquationId();
+      rResult[local_index++] = this->GetGeometry()[i_node].GetDof(DISPLACEMENT_Z, d_pos + 2).EquationId();
+  }
+
+  KRATOS_CATCH("")
+}
+
+//***********************************************************************************
+//***********************************************************************************
+
+void MembraneElement::GetDofList(
+    DofsVectorType& rElementalDofList,
+    ProcessInfo& rCurrentProcessInfo)
+
+{
+    SizeType num_nodes, local_size;
+    num_nodes = GetGeometry().size();
+    local_size = num_nodes * 3;
+
+    if (rElementalDofList.size() != local_size)
+        rElementalDofList.resize(local_size);
+
+    SizeType local_index = 0;
+
+    for (SizeType i_node = 0; i_node < num_nodes; ++i_node)
+    {
+        rElementalDofList[local_index++] = this->GetGeometry()[i_node].pGetDof(DISPLACEMENT_X);
```

Comment:
use the version with position, it should be faster

Reply:
`GetDof` has a version with position, `pGetDof` does not